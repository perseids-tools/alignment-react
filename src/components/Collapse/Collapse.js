import React, { useState, useEffect } from 'react';
import { bool, node, string } from 'prop-types';

import styles from './Collapse.module.scss';

const Collapse = ({ title, children, collapsed: beginCollapsed }) => {
  const [collapsed, setCollapsed] = useState(beginCollapsed);
  const toggleCollapse = () => setCollapsed(!collapsed);

  useEffect(() => {
    setCollapsed(beginCollapsed);
  }, [beginCollapsed]);

  return (
    <div>
      <button type="button" className={styles.button} onClick={toggleCollapse}>
        <div>
          {title}
        </div>
        <div className={styles.right}>
          {collapsed ? '+' : '-'}
        </div>
      </button>
      {collapsed ? null : children}
    </div>
  );
};

Collapse.propTypes = {
  title: string.isRequired,
  children: node.isRequired,
  collapsed: bool,
};

Collapse.defaultProps = {
  collapsed: true,
};

export default Collapse;
