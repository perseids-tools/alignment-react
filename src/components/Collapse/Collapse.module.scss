.button {
  background-color: #8f8f8f;
  color: #ffffff;
  font-size: 1rem;
  width: 100%;

  display: flex;
  justify-content: space-between;
  border: none;
  cursor: pointer;

  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;

  margin-bottom: 0.2rem;
}

.button:hover {
  background-color: #707070;
}

.right {
  font-weight: bold;
}
