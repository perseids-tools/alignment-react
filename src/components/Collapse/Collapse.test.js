import React from 'react';
import renderer from 'react-test-renderer';

import Collapse from './Collapse';

it('renders collapsed', () => {
  const component = (
    <Collapse title="Title">
      <div>Hello world</div>
    </Collapse>
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders uncollapsed', () => {
  const component = (
    <Collapse title="Title" collapsed={false}>
      <div>Hello world</div>
    </Collapse>
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});
