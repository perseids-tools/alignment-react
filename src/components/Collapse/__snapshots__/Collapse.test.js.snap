// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders collapsed 1`] = `
<div>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    <div>
      Title
    </div>
    <div
      className="right"
    >
      +
    </div>
  </button>
</div>
`;

exports[`renders uncollapsed 1`] = `
<div>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    <div>
      Title
    </div>
    <div
      className="right"
    >
      -
    </div>
  </button>
  <div>
    Hello world
  </div>
</div>
`;
